src/services/item/plugins/html/h5p/index.ts

```diff
@@ -0,0 +1,225 @@
+
+  // question: this is difficult to move this in the service because of the transaction
+  /**
+   * Creates a Graasp item for the uploaded H5P package
+   * @param filename Name of the original H5P file WITHOUT EXTENSION
+   * @param contentId Storage ID of the remote content
+   * @param remoteRootPath Root path on the remote storage
+   * @param member Actor member
+   * @param parentId Optional parent id of the newly created item
+   */
+  async function createH5PItem(
+    member: Member,
+    filename: string,
+    contentId: string,
+    parentId?: string,
+  ): Promise<Item> {
+    const metadata = {
+      name: h5pService.buildH5PPath('', filename),
+      type: ItemType.H5P,
+      extra: h5pService.buildH5PExtra(contentId, filename),
+    };
+    return db.transaction(async (manager) => {
+      return itemService.post(member, buildRepositories(manager), {
+        item: metadata,
+        parentId,
+      });
+    });
+  }
+
+  /**
+   * In local storage mode, proxy serve the H5P content files
+   * In the future, consider refactoring the fileService so that it can be grabbed from the
+   * core instance and can serve the files directly (with an option to use or not auth)
+   */
+  if (fileStorage.type === ItemType.LOCAL_FILE) {
+    /** Helper to set CORS headers policy */
+    const setHeaders = (response: FastifyStaticReply) => {
+      response.setHeader('Cross-Origin-Resource-Policy', 'same-site');
+    };
+
+    // serve integration html
+    const integrationRoute = path.join(
+      routes?.assets ?? DEFAULT_H5P_ASSETS_ROUTE,
+      'integration.html',
+    );
+    fastify.get(integrationRoute, async (req, res) => {
+      const html = renderHtml(
+        routes?.assets ?? DEFAULT_H5P_ASSETS_ROUTE,
+        routes?.content ?? DEFAULT_H5P_CONTENT_ROUTE,
+        hosts ?? ['localhost'],
+      );
+      res.send(html);
+    });
+
+    // hack to serve the "dist" folder of package "h5p-standalone"
+    const h5pAssetsRoot = path.dirname(require.resolve('h5p-standalone'));
+    fastify.register(fastifyStatic, {
+      root: h5pAssetsRoot,
+      prefix: routes?.assets ?? DEFAULT_H5P_ASSETS_ROUTE,
+      decorateReply: false,
+      setHeaders,
+    });
+
+    const h5pStorageRoot = path.join(fileStorage.config.local.storageRootPath, pathPrefix);
+    fs.mkdirSync(h5pStorageRoot, { recursive: true });
+    fastify.register(fastifyStatic, {
+      root: h5pStorageRoot,
+      prefix: routes?.content ?? DEFAULT_H5P_CONTENT_ROUTE,
+      decorateReply: false,
+      setHeaders,
+    });
+  }
+
+  /*
+    we create an artificial plugin scope, so that fastify-multipart does not conflict
+    with other instances since we use fp to remove the outer scope
+  */
+  await fastify.register(async (fastify) => {
+    fastify.register(fastifyMultipart, {
+      limits: {
+        fields: MAX_NON_FILE_FIELDS,
+        files: MAX_FILES,
+        fileSize: MAX_FILE_SIZE,
+      },
+    });
```

Comment:
Arguably this could be factored out as well. But probably not in the html service anyway